daemon/containerd/image_delete.go

```diff
@@ -108,6 +108,9 @@ func (i *ImageService) ImageDelete(ctx context.Context, imageRef string, force,
 			return nil, err
 		}
 
+		if isDanglingImage(img) {
+			return nil, nil
+		}
```

Comment:
If we can already know beforehand that `img` is a dangling image, why are we still calling `softImageDelete` on it? Is `i.softImageDelete` robust to being passed a dangling image? From a read of the sources, it looks like the answer is no: it will delete the dangling image.

Reply:
This isn't really something that should happen in "normal scenario" as a dangling image should be picked up earlier and wouldn't reach this point at all.
Currently it can only reach here if you use an explicit dangling name (`moby-dangling@digest`), but we don't really want users to actually use this name.
Removed this check and adjusted the earlier code to handle this situation earlier.